src/deployment/deploy.py

```diff
@@ -305,7 +304,7 @@ def get_instance_url(self) -> str:
         # The url to access the instance
         # This also represents the legacy identifier_uris of the application
         # registration
-        if self.multi_tenant_domain:
+        if self.multi_tenant_domain != "":
```

Comment:
if not self.multi_tenant_domain:  <<-- will be true if not null and not empty

https://docs.python.org/3/library/stdtypes.html#truth-value-testing